lib/mayaUsd/fileio/translators/translatorXformable.cpp

```diff
@@ -201,6 +267,61 @@ static void _setMayaAttribute(
     }
 }
 
+static void _applyToMayaAttribute(
+    MFnDagNode&                     depFn,
+    const std::vector<double>&      xVal,
+    const std::vector<double>&      yVal,
+    const std::vector<double>&      zVal,
+    const MTimeArray&               timeArray,
+    const MString&                  opName,
+    const MString&                  x,
+    const MString&                  y,
+    const MString&                  z,
+    const UsdMayaPrimReaderContext* context)
+{
+    MPlug plg;
+    if (x != "" && !xVal.empty()) {
+        plg = depFn.findPlug(opName + x);
+        if (!plg.isNull()) {
+            double curValue = plg.asDouble();
+            if (opName == "scale") {
+                plg.setDouble(xVal[0] * curValue);
+            } else {
+                plg.setDouble(xVal[0] + curValue);
+            }
+            if (xVal.size() > 1 && _isArrayVarying(xVal))
+                _applyAnimPlugData(plg, xVal, timeArray, context);
+        }
+    }
+    if (y != "" && !yVal.empty()) {
+        plg = depFn.findPlug(opName + y);
+        if (!plg.isNull()) {
+            double curValue = plg.asDouble();
+            if (opName == "scale") {
+                plg.setDouble(yVal[0] * curValue);
+            } else {
+                plg.setDouble(yVal[0] + curValue);
+            }
+            if (yVal.size() > 1 && _isArrayVarying(yVal))
+                _applyAnimPlugData(plg, yVal, timeArray, context);
+        }
+    }
+    if (z != "" && !zVal.empty()) {
+        plg = depFn.findPlug(opName + z);
+        if (!plg.isNull()) {
+            double curValue = plg.asDouble();
+            if (opName == "scale") {
+                plg.setDouble(zVal[0] * curValue);
+            } else {
+                plg.setDouble(zVal[0] + curValue);
+            }
+            if (zVal.size() > 1 && _isArrayVarying(zVal))
+                _applyAnimPlugData(plg, zVal, timeArray, context);
+        }
+    }
+    return;
+}
```

Comment:
Your new _applyToMayaAttribute() and _setMayaAttribute() has very similar function signature with minor changes in logics. I would consolidate these two routines and pass a flag to separate the logics.